comp/core/workloadmeta/collectors/util/image_util.go

```diff
@@ -56,11 +104,20 @@ func removeOldRepoProperties(properties []cyclonedx.Property) []cyclonedx.Proper
 	return res
 }
 
-// appendRepoProperties function updates the slice of properties
-func appendRepoProperties(properties []cyclonedx.Property, repoValues []string, propertyKeyType string) []cyclonedx.Property {
-	for _, repoValue := range repoValues {
-		prop := cdxProperty(propertyKeyType, repoValue)
-		properties = append(properties, prop)
+// Append new values to the properties that were not already present
+func appendProperties(properties []cyclonedx.Property, newValues []string, propertyKeyType string) []cyclonedx.Property {
+	existingValues := make(map[string]bool)
+	for _, prop := range properties {
+		if prop.Name == propertyKeyType {
+			existingValues[prop.Value] = true
```

Comment:
For implementing sets, the usual GO pattern is to use `map`s of `struct{}`.

```suggestion
	existingValues := make(map[string]struct{})
	for _, prop := range properties {
		if prop.Name == propertyKeyType {
			existingValues[prop.Value] = struct{}{}
```